haigha/transports/socket_transport.py

```diff
@@ -29,23 +29,33 @@ def connect(self, (host, port), klass=socket.socket):
         Connect assuming a host and port tuple.
         '''
         self._host = "%s:%s" % (host, port)
-        self._sock = klass()
+        infos = socket.getaddrinfo(host, port, 0, 0, socket.IPPROTO_TCP)
+        if not infos:
+            raise socket.error("getaddrinfo returns an empty list")
+
+        # Sort addresses such that the new IPv6 address is first.
+        for info in reversed(sorted(infos, key=lambda info: len(info[-1]))):
```

Comment:
I missed that section of the docs talking about the order of `getaddrinfo`. I'm happy you caught that. Originally I was prioritizing IPv6 simply because it is the newer protocol. I'll definitely need to remove the reordering.